crates/air-lib/interpreter-signatures/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#![forbid(unsafe_code)]
+#![warn(rust_2018_idioms)]
+#![deny(
+    dead_code,
+    nonstandard_style,
+    unused_imports,
+    unused_mut,
+    unused_variables,
+    unused_unsafe,
+    unreachable_patterns
+)]
+
+use air_interpreter_cid::CID;
+use fluence_keypair::error::SigningError;
+use fluence_keypair::KeyPair;
+use rand_chacha::rand_core::SeedableRng;
+use serde::{Deserialize, Serialize};
+
+use std::borrow::Borrow;
+use std::collections::HashMap;
+use std::hash::Hash;
+
+/// An opaque serializable representation of public key.
+///
+/// It can be string or binary, you shouldn't care about it unless you change serialization format.
+#[derive(Debug, Hash, Clone, Eq, PartialEq, PartialOrd, Deserialize, Serialize)]
+#[serde(transparent)]
+pub struct PublicKey(Box<str>);
+
+impl From<fluence_keypair::PublicKey> for PublicKey {
+    fn from(value: fluence_keypair::PublicKey) -> Self {
+        Self(bs58::encode(&value.to_vec()).into_string().into())
+    }
+}
+
+#[derive(Debug, Hash, Clone, Eq, PartialEq, PartialOrd, Deserialize, Serialize)]
+#[serde(transparent)]
+pub struct Signature(Box<str>);
```

Comment:
ditto

Reply:
As we use JSON yet, a string is more effective.  The type is designed to be opaque, and can be easily changed to `Vec<u8>` (or `Box<[u8]>`) when we switch to binary format.